lib/core/task.toit

```diff
@@ -75,6 +76,90 @@ interface Task:
   static current -> Task:
     return Task_.current
 
+  /**
+  Divides the work given by $lambdas up into a number of
+    separate tasks that run concurrently.
+
+  Returns the results of running the $lambdas as a $Map
+    keyed by the lambda index. If the lambda at a given
+    index did not return, the results map will not contain
+    and entry for it. The results map is insertion ordered,
+    so it is possible to tell which lambda returned first.
+
+  If any of $lambdas throw an exception, the exception is
+    propagated to the caller of $Task.divide which in
+    returns also throws.
+
+  If $required is less than the number of $lambdas, the
+    method returns when $required tasks have completed.
+  */
+  static divide lambdas/List -> Map
```

Comment:
The main issue I have with this API is that it is hard to give additional properties to the individual lambdas.

I could imagine that the following related scenarios would be useful:
- run in parallel, and restart tasks that fail
- ignore failed tasks completely
- trace failed tasks, but don't stop the others

If we didn't provide a lambda, but an instance, then we could add these flags to them:
```
Task.run_in_parallel [
  XX --ignore_failure:: do_something,
  XX --keep_running --trace_errors:: ...,
]
```

Maybe, if we decide to adopt `|>`:
```
results := Task.in_parallel
|> %.run --keep_running:: do_something
|> %.run --ignore_error:: do_something_else
|> %.run ...
|> %.wait_for_results
```

The latter could already be implemented now:
```
scheduler := Task.in_parallel
scheduler.run --ignore_failure:: ...
scheduler.run --keep_running:: ...
results := scheduler.wait_for_results
```

Looking at it, we could probably always add the more advanced version later, as it doesn't have the same function shape as this implementation.

Reply:
The three suggestions are all achievable (quite easily) from within the individual lambdas. It is essentially just variants of `catch` used in them.